workers/unity/Assets/Tests/EditmodeTests/Correctness/Subscriptions/RequireablesReenableTests.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.Commands;
+using Improbable.Gdk.PlayerLifecycle;
+using Improbable.Gdk.Subscriptions;
+using Improbable.Gdk.Test;
+using Improbable.Gdk.TestUtils;
+using Improbable.Worker.CInterop;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Empty = Improbable.Gdk.Core.Empty;
+using Entity = Unity.Entities.Entity;
+
+namespace Improbable.Gdk.EditmodeTests
+{
+    [TestFixture]
+    public class RequireablesReenableTests : MockBase
+    {
+        private const long EntityId = 100;
+
+        [Test]
+        public void CommandSenders_and_Receivers_are_valid_after_reinjection()
+        {
+            World
+                .Step(world => world.Connection.CreateEntity(EntityId, GetTemplate()))
+                .Step(world =>
+                    world.Connection.ChangeAuthority(EntityId, TestCommands.ComponentId, Authority.Authoritative))
+                .Step(world =>
+                {
+                    var (_, behaviour) = world.CreateGameObject<CommandSenderReceiverTestBehaviour>(EntityId);
+                    return behaviour;
+                })
+                .Step((world, behaviour) =>
+                {
+                    Assert.IsNotNull(behaviour.CommandReceiver);
+                    Assert.IsNotNull(behaviour.CommandSender);
+                    Assert.IsTrue(behaviour.CommandReceiver.IsValid);
+                    Assert.IsTrue(behaviour.CommandSender.IsValid);
+                })
+                .Step(world =>
+                    world.Connection.ChangeAuthority(EntityId, TestCommands.ComponentId, Authority.NotAuthoritative))
+                .Step((world, behaviour) =>
+                {
+                    Assert.IsNull(behaviour.CommandReceiver);
+                    Assert.IsNull(behaviour.CommandSender);
```

Comment:
Should this check if the captured command sender/receiver is invalidated?

Reply:
Those are captured in https://github.com/spatialos/gdk-for-unity/blob/develop/workers/unity/Assets/Tests/EditmodeTests/Correctness/Subscriptions/RequireablesDisableTests.cs :)